### PR TITLE
feat(aws-efs): grant support on FileSystem

### LINKED ISSUE #14998: (aws-efs): Add fileSystemArn property and grant method for FileSystem construct
When creating or importing a File System in a CDK App, the construct is missing the `fileSystemArn` property.

### Use Case

When trying to grant permissions to a FileSystem, we have to build the fileSystem the policy and the arn manually.

```ts
const customStack = new Stack(app);

const customVpc = new ec2.Vpc(customStack, 'VPC');
const fs = new FileSystem(customVpc, 'EfsFileSystem', {
  vpc: customVpc,
});

new iam.PolicyStatement({
  actions: ['elasticfilesystem:ClientWrite'],
  resources: [Stack.of(scope).formatArn({
    service: 'elasticfilesystem',
    resource: 'file-system',
    resourceName: fs.fileSystemId,
  })],
});
```

That would become

```ts
const customStack = new Stack(app);

const customVpc = new ec2.Vpc(customStack, 'VPC');
const fs = new FileSystem(customVpc, 'EfsFileSystem', {
  vpc: customVpc,
});

fs.grant(grantee, 'elasticfilesystem:ClientWrite');
```

### Proposed Solution

Add a `fileSystemArn` attribute to the construct.
Allow import using arn.
Add a grant method.

* [x] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by DaWyz**
`FileSystem` now have the `fileSystemArn` attribute available. It is also possible to import an existing `FileSystem` by `arn` or `id` using the `fromFileSystemAttributes` method.

You can also grant permission to an existing grantee using the grant method.

See the example below giving an IAM Role permission to write to an imported file system:

```ts
const arn = stack.formatArn({
  service: 'elasticfilesystem',
  resource: 'file-system',
  resourceName: 'fs-12912923',
});

const importedFileSystem = efs.FileSystem.fromFileSystemAttributes(this, 'existingFS', {
  fileSystemArn: arn, // You can also use fileSystemArn instead of fileSystemId.
  securityGroup: ec2.SecurityGroup.fromSecurityGroupId(this, 'SG', 'sg-123456789', {
    allowAllOutbound: false,
  }),
});

const role = new iam.Role(this, 'Access Role', { assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com') });

importedFileSystem.grant(role, 'elasticfilesystem:ClientWrite');
```

Closes #14998.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-efs/README.md [hunk 1]
```diff
@@ -37,12 +37,33 @@ const fileSystem = new efs.FileSystem(this, 'MyEfsFileSystem', {
...
+  securityGroup: ec2.SecurityGroup.fromSecurityGroupId(this, 'SG', 'sg-123456789', {
+    allowAllOutbound: false,
+  }),
+});
+
+const role = new iam.Role(this, 'Role', {
+  assumedBy: new iam.AnyPrincipal(),
+});
+
+importedFileSystem.grant(role, 'elasticfilesystem:ClientWrite');
```
Comment (nija-at) [inline packages/@aws-cdk/aws-efs/README.md:64]: Document the grant API in a separate section.

**Review by nija-at — DISMISSED** [review]
Thanks for the PR @DaWyz. Looks mostly good. A few comments below.

File: packages/@aws-cdk/aws-efs/lib/efs-file-system.ts [hunk 2]
```diff
@@ -271,6 +333,11 @@ export class FileSystem extends Resource implements IFileSystem {
     filesystem.applyRemovalPolicy(props.removalPolicy);
 
     this.fileSystemId = filesystem.ref;
+    this.fileSystemArn = Stack.of(scope).formatArn({
+      service: 'elasticfilesystem',
+      resource: 'file-system',
+      resourceName: this.fileSystemId,
+    });
```
[the remark this answers is not in the document]
Comment (DaWyz) [reply packages/@aws-cdk/aws-efs/lib/efs-file-system.ts]: Good catch. Just updated the code.

File: packages/@aws-cdk/aws-efs/lib/efs-file-system.ts [hunk 3]
```diff
@@ -330,7 +402,23 @@ class ImportedFileSystem extends Resource implements IFileSystem {
   constructor(scope: Construct, id: string, attrs: FileSystemAttributes) {
     super(scope, id);
 
-    this.fileSystemId = attrs.fileSystemId;
+    if (!attrs.fileSystemId && !attrs.fileSystemArn) {
```
[the remark this answers is not in the document]
Comment (DaWyz) [reply packages/@aws-cdk/aws-efs/lib/efs-file-system.ts:402]: I'm going to throw an error if both are provided (similar to how it's done for DynamoDB tables).

File: packages/@aws-cdk/aws-efs/lib/efs-file-system.ts [hunk 4]
```diff
@@ -330,7 +399,27 @@ class ImportedFileSystem extends Resource implements IFileSystem {
...
+      throw new Error('One of fileSystemId or fileSystemArn must be provided.');
+    }
+
+    if (attrs.fileSystemId && attrs.fileSystemArn) {
+      throw new Error('Only one of fileSystemId or fileSystemArn must be provided.');
+    }
```
Comment (nija-at) [inline packages/@aws-cdk/aws-efs/lib/efs-file-system.ts:408]: Add unit tests for these

**Review by nija-at — CHANGES_REQUESTED** [review]
Thanks for making the changes. A few more comments below.

**Comment by DaWyz** [comment]
Updates the code following last recommendations. Also updated the pull request description.

**Review by nija-at — APPROVED** [review]
Awesome! Looks great.

**Review by nija-at — CHANGES_REQUESTED** [review]
@DaWyz - can you take a look at the failing build with the latest changes.

It's likely because you need to update the tests to use the new assertion framework - https://github.com/aws/aws-cdk/commit/367fe7e1e05be31b77fbf7c6e2e60e126e854a59

**Review by nija-at — APPROVED** [review]

**Pull request merged**